### ISSUE TITLE
teamcity tests can fail tests yet exit 0 and report as passed

### ISSUE THREAD

**Issue by jrgm** [mozilla/fxa-content-server#2795]
@vladikoff - I noticed this by chance but look at 

http://tests.dev.lcip.org:8111/viewLog.html?buildId=4357&buildTypeId=FxaLatest_FunctionalTests&tab=buildLog#_focus=18&state=18

At the end it says, 
```
[07:26:57] [38;5;10m✓ firefox on any platform - oauth permissions - signup, verify same browser [37m (12.287s) [0m
[07:27:02] [38;5;10m✓ firefox on any platform - oauth permissions - preverified sign up [37m (4.096s) [0m
[07:27:20] [38;5;10m✓ firefox on any platform - oauth permissions - signup, then sign in [37m (15.387s) [0m
[07:27:20]No unit test coverage for firefox on any platform [0m
[07:27:20] [38;5;9mfirefox on any platform: 1/338 tests failed [0m
[07:27:20]
[07:27:20]
[07:27:20] [1m [41mTOTAL: tested 1 platforms, 1/338 tests failed [0m
[07:27:20]Process exited with code 0
[07:27:20]Publishing internal artifacts
[07:27:20]Build finished
```

**Comment by jrgm** [comment]
Oh, I see. `latest` still runs a set of commands in the text box, not the ./run.sh script. 

That set of commands has last command of `rm -rf ~/fxatemp` so "exit 0".

I can just put `set -e` in that set of commands to exit on first error.

**Comment by vladikoff** [comment]
> I can just put set -e in that set of commands to exit on first error.

Please! and thanks for noticing the issue, makes no sense why it does not error out when at least one process fails :(

**Comment by jrgm** [comment]
That's what either `set -e` or `&&` do ;-)

**Comment by jrgm** [comment]
Okay, this time with `set -e` the run ends with the below and reports as failed. Yaaay!!, um, oh...

[21:24:05] [38;5;10m✓ firefox on any platform - oauth permissions - signup, verify same browser [37m (13.076s) [0m
[21:24:11] [38;5;10m✓ firefox on any platform - oauth permissions - preverified sign up [37m (4.983s) [0m
[21:24:30] [38;5;10m✓ firefox on any platform - oauth permissions - signup, then sign in [37m (17.053s) [0m
[21:24:30]No unit test coverage for firefox on any platform [0m
[21:24:30] [38;5;9mfirefox on any platform: 1/338 tests failed [0m
[21:24:30]
[21:24:30]
[21:24:30] [1m [41mTOTAL: tested 1 platforms, 1/338 tests failed [0m
[21:24:30]Process exited with code 1
[21:24:30]Step Run Tests (Command Line) failed
[21:24:30]Publishing internal artifacts
[21:24:31]Build finished

**Comment by jrgm** [comment]
The failed test is 
```
[21:15:08] [38;5;10m✓ firefox on any platform - oauth sign up verification_redirect - signup, same browser different window, verification_redirect=always [37m (9.024s) [0m
[21:15:18] [38;5;10m✓ firefox on any platform - oauth sign up verification_redirect - signup, verify different browser, verification_redirect=always [37m (8.039s) [0m
[21:15:57] [38;5;9m× firefox on any platform - oauth reset password - reset password, verify same browser [37m (36.967s)
[21:15:57] [31mNoSuchElement: [POST http://localhost:4444/wd/hub/session/426754d0-64cf-49b0-acc9-fe4562cc5a2e/element / {"using":"css selector","value":"#loggedin"}] Unable to locate element: {"method":"css selector","selector":"#loggedin"}
[21:15:57]For documentation on this error, please visit: http://seleniumhq.org/exceptions/no_such_element.html
[21:15:57]Build info: version: '2.46.0', revision: '87c69e2', time: '2015-06-04 16:16:47'
[21:15:57]System info: host: 'ip-172-31-11-113', ip: '172.31.11.113', os.name: 'Linux', os.arch: 'amd64', os.version: '3.13.0-51-generic', java.version: '1.7.0_80'
[21:15:57]Driver info: driver.version: unknown
[21:15:57]  at ProxiedSession._post  <node_modules/intern/node_modules/leadfoot/Session.js:59:30>
[21:15:57]  at ProxiedSession.Session.find  <node_modules/intern/node_modules/leadfoot/Session.js:1055:15>
[21:15:57]  at Command.<anonymous>  <node_modules/intern/node_modules/leadfoot/Command.js:42:36>
[21:15:57]  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
[21:15:57]  at Object.run [as _onImmediate]  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
[21:15:57]  at processImmediate [as _immediateCallback]  <timers.js:354:15>
[21:15:57]  at Command.find  <node_modules/intern/node_modules/leadfoot/Command.js:23:10>
[21:15:57]  at Command.prototype.(anonymous function) [as findByCssSelector]  <node_modules/intern/node_modules/leadfoot/lib/strategies.js:24:16>
[21:15:57]  at Test.registerSuite.reset password, verify same browser [as test]  <tests/functional/oauth_reset_password.js:118:10>
[21:15:57]  at start.then.finally.self.timeElapsed  <node_modules/intern/lib/Test.js:211:24>
[21:15:57]  at <node_modules/intern/node_modules/dojo/Promise.ts:393:15>
[21:15:57]  at runCallbacks  <node_modules/intern/node_modules/dojo/Promise.ts:11:11>
[21:15:57]  at <node_modules/intern/node_modules/dojo/Promise.ts:317:4>
[21:15:57]  at Object.run [as _onImmediate]  <node_modules/intern/node_modules/dojo/Promise.ts:237:7>
[21:15:57]  at processImmediate [as _immediateCallback]  <timers.js:354:15> [0m
[21:16:06] [38;5;10m✓ firefox on any platform - oauth reset password - reset password, verify same browser with original tab closed [37m (8.099s) [0m
```
